File: LeetCode/0020. Valid Parentheses/JavaScript/01. Valid Parentheses.js
var isValid = function (s) {
    let stackBracket = [];
    for (let bracketChar of s.split("")) {
        switch (bracketChar) {
            case "(":
            case "[":
            case "{":
                stackBracket.push(bracketChar);
                break;
            case ")":
                if (stackBracket.length === 0 || stackBracket.pop() != "(") {
                    return false;
                }
                break;
            case "]":
                if (stackBracket.length === 0 || stackBracket.pop() != "[") {
                    return false;
                }
                break;
            case "}":
                if (stackBracket.length === 0 || stackBracket.pop() != "{") {
                    return false;
                }
                break;
        }
    }
    return stackBracket.length === 0;
};

File: LeetCode/0020. Valid Parentheses/C#/01. Valid Parentheses.cs
bool IsValid(string s)
{
    Stack<char> stackBracket = new Stack<char>();
    foreach (char bracketChar in s)
    {
        switch (bracketChar)
        {
            case '(':
            case '[':
            case '{':
                stackBracket.Push(bracketChar);
                break;
            case ')':
                if (stackBracket.Count == 0 || stackBracket.Pop() != '(')
                {
                    return false;
                }
                break;
            case ']':
                if (stackBracket.Count == 0 || stackBracket.Pop() != '[')
                {
                    return false;
                }
                break;
            case '}':
                if (stackBracket.Count == 0 || stackBracket.Pop() != '{')
                {
                    return false;
                }
                break;
        }
    }
    return stackBracket.Count == 0;
}